boards/components/src/sched/mlfq.rs

```diff
@@ -0,0 +1,99 @@
+//! Component for a multi-level feedback queue scheduler.
+//!
+//! This provides one Component, MLFQComponent.
+//!
+//! Usage
+//! -----
+//! (Must be final lines in main.rs)
+//! ```rust
+//! let scheduler =
+//!     components::mlfq::MLFQComponent::new(board_kernel, mux_alarm, &PROCESSES).finalize(
+//!         components::mlfq_component_helper!(sam4l::ast::Ast, NUM_PROCS),
+//!     );
+//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap);
```

Comment:
```suggestion
//! Note that `kernel_loop` never returns, thus this must end main.rs
//! ```rust
//! let scheduler =
//!     components::mlfq::MLFQComponent::new(board_kernel, mux_alarm, &PROCESSES).finalize(
//!         components::mlfq_component_helper!(sam4l::ast::Ast, NUM_PROCS),
//!     );
//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap)
```